sormas-backend/src/main/java/de/symeda/sormas/backend/event/eventimport/EventImportFacadeEjb.java

```diff
@@ -0,0 +1,615 @@
+import de.symeda.sormas.api.user.UserDto;
+import de.symeda.sormas.api.user.UserReferenceDto;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.api.utils.DateHelper;
+import de.symeda.sormas.api.utils.ValidationRuntimeException;
+import de.symeda.sormas.backend.event.EventFacadeEjb.EventFacadeEjbLocal;
+import de.symeda.sormas.backend.event.EventParticipantFacadeEjb.EventParticipantFacadeEjbLocal;
+import de.symeda.sormas.backend.facility.FacilityFacadeEjb.FacilityFacadeEjbLocal;
+import de.symeda.sormas.backend.person.PersonFacadeEjb.PersonFacadeEjbLocal;
+import de.symeda.sormas.backend.region.AreaFacadeEjb.AreaFacadeEjbLocal;
+import de.symeda.sormas.backend.region.DistrictFacadeEjb.DistrictFacadeEjbLocal;
+import de.symeda.sormas.backend.region.RegionFacadeEjb.RegionFacadeEjbLocal;
+import de.symeda.sormas.backend.user.UserFacadeEjb.UserFacadeEjbLocal;
+import de.symeda.sormas.backend.user.UserService;
+
+@Stateless(name = "EventImportFacade")
+public class EventImportFacadeEjb implements EventImportFacade {
+
+	private final Logger LOGGER = LoggerFactory.getLogger(EventImportFacadeEjb.class);
+
+	protected static final String ERROR_COLUMN_NAME = I18nProperties.getCaption(Captions.importErrorDescription);
+
+	@EJB
+	private UserService userService;
+	@EJB
+	private PersonFacadeEjbLocal personFacade;
+	@EJB
+	private EventFacadeEjbLocal eventFacade;
+	@EJB
+	private EventParticipantFacadeEjbLocal eventParticipantFacade;
+	@EJB
+	private UserFacadeEjbLocal userFacade;
+	@EJB
+	private AreaFacadeEjbLocal areaFacade;
+	@EJB
+	private DistrictFacadeEjbLocal districtFacade;
+	@EJB
+	private RegionFacadeEjbLocal regionFacade;
+	@EJB
+	private CommunityFacadeEjbLocal communityFacade;
+	@EJB
+	private FacilityFacadeEjbLocal facilityFacade;
+
+	@Override
+	@Transactional
+	public ImportLineResultDto<EventImportEntities> importEventData(
+		String[] values,
+		String[] entityClasses,
+		String[] entityProperties,
+		String[][] entityPropertyPaths,
+		boolean ignoreEmptyEntries)
+		throws InvalidColumnException {
+
+		// Check whether the new line has the same length as the header line
+		if (values.length > entityProperties.length) {
+			return ImportLineResultDto.errorResult(I18nProperties.getValidationError(Validations.importLineTooLong));
+		}
+
+		final EventImportEntities entities = new EventImportEntities(userService.getCurrentUser().toReference());
+		ImportLineResultDto<EventImportEntities> importResult =
+			buildEntities(values, entityClasses, entityPropertyPaths, ignoreEmptyEntries, entities);
+		if (importResult.isError()) {
+			return importResult;
+		}
+
+		ImportLineResultDto<EventImportEntities> validationResult = validateEntities(entities);
+		if (validationResult.isError()) {
+			return validationResult;
+		}
+
+		for (EventParticipantDto eventParticipant : entities.getEventParticipants()) {
+			PersonDto person = eventParticipant.getPerson();
+
+			if (isPersonSimilarToExisting(person)) {
```

Comment:
Will this work when there are two or more event participants in one line where there's a similar person in the database? Or will only the first person be checked and all others created without any further user input?

Reply:
If a row contains several event participant that are already present in the system, the merge popup will appear for each event participant to merge.
This process will be repeated for each row.